### PR TITLE
Add Tooltip component

### LINKED ISSUE #42: Tooltips
- We need to think about what scenarios can trigger what patterns.
  - Info Icon triggers tooltip
  - Text link triggers modal

- [x] Bubble
- [x] Text
- [x] Caret in all directions

### PR DESCRIPTION
**Pull request by cristianoliveira**
Changes:
  - Add tooltip component with optional alignment of `center`, `right` or `left`.

closes #42

### CODE REVIEW

File: src/components/Tooltip/Tooltip.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    top: 100%;
+    ${arrowAligns(theme)[align]};
+  }
+`;
+
+const TooltipText = styled('span')(tooltipTextStyles);
+const TooltipElement = styled('div')(baseStyles);
+
+/**
+ * A Tooltip wrapper in order to show tooltips ontop of other
```
Comment (mlent) [inline src/components/Tooltip/Tooltip.js]: ontop -> on top

File: src/components/Tooltip/Tooltip.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+ * components.
+ */
+const Tooltip = ({ children, content, align, ...props }) => {
+  return (
+    <TooltipElement {...props}>
+      <TooltipText {...{ align }}>{content}</TooltipText>
+      {children}
+    </TooltipElement>
+  );
+};
```
Comment (mlent) [inline src/components/Tooltip/Tooltip.js]: You can omit the `{ return (` and just do `(` :)

File: src/components/Tooltip/Tooltip.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+Tooltip.Center = 'center';
+Tooltip.Left = 'left';
+Tooltip.Right = 'right';
+
+Tooltip.propTypes = {
+  /**
+   * The content inside of the tooltip being show.
```
Comment (mlent) [inline src/components/Tooltip/Tooltip.js]: `show` -> `shown`

File: src/components/Tooltip/Tooltip.js [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+   */
+  content: PropTypes.node.isRequired,
```
Comment (mlent) [inline src/components/Tooltip/Tooltip.js]: Could these be node or string?

Comment (cristianoliveira) [reply src/components/Tooltip/Tooltip.js]: Actually, I didn't want to fix a type of this. On examples I we only have `text` but isn't some icon also acceptable too?

File: src/components/Tooltip/Tooltip.js [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    right: 0;
+    margin-left: auto;
+    margin-right: auto;
+  `,
+  right: 'left: 50%;',
+  left: 'right: 50%;'
+};
+
+const tooltipTextStyles = ({ align, theme }) => css`
+  label: tooltip_text;
```
Comment (mlent) [inline src/components/Tooltip/Tooltip.js]: Don't you need to have something like `label: tooltip__text` and then for the alignment, `label: tooltip__text--${align}`?

File: src/components/Tooltip/Tooltip.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+  border-radius: ${theme.borderRadius.mega};
+  padding: ${theme.spacings.byte};
+  position: absolute;
+  z-index: 1;
+  bottom: 125%;
+  ${tooltipAlign[align]};
+  transition: opacity 0.3s;
+  ${textKilo({ theme })};
+
+  &:after {
```
Comment (felixjung) [inline src/components/Tooltip/Tooltip.js]: Please use `::` for `before` and `after`.

File: src/components/Tooltip/Tooltip.js [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+      opacity: 1;
+    }
+  }
+`;
+
+const arrowAligns = theme => ({
+  center: `left: 50%`,
+  right: `left: ${theme.spacings.mega}`,
+  left: `right: ${theme.spacings.mega}`
+});
```
Comment (connor-baer) [inline src/components/Tooltip/Tooltip.js]: Although it's a neat solution, this won't work with our static style extraction.

File: src/components/Tooltip/Tooltip.spec.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+
+import Tooltip from '.';
+
+describe('Tooltip', () => {
+  it('aligns to the center', () => {
```
Comment (felixjung) [inline src/components/Tooltip/Tooltip.spec.js]: I phrased it like this, too 😄 But we agreed to use `should...` in the team.

File: src/components/Tooltip/Tooltip.js [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+ * components.
+ */
+const Tooltip = ({ children, content, align, ...props }) => (
+  <TooltipElement {...props}>
+    <TooltipContainer {...{ align }}>{content}</TooltipContainer>
+    {children}
+  </TooltipElement>
+);
+
+Tooltip.Center = 'center';
```
Comment (felixjung) [inline src/components/Tooltip/Tooltip.js]: Are we making these constants instead? I think that's what I've done in the others. For example, `CENTER`, `LEFT`, `RIGHT`. Maybe using them in a more enum style like you did here. What do you all think?

File: src/components/Tooltip/Tooltip.spec.js [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+    const component = create(
+      <Tooltip align={Tooltip.Center} content="The tooltip content">
+        Something with tooltip
+      </Tooltip>
+    ).toJSON();
```
Comment (felixjung) [inline src/components/Tooltip/Tooltip.spec.js]: I think with the emotion serializer, we don't need to call `toJSON()`. It takes care of this for us.

File: src/components/Tooltip/Tooltip.spec.js (hunk 8 above)
Comment (cristianoliveira) [reply src/components/Tooltip/Tooltip.spec.js]: Ops. habits :) 
Let me change

File: src/components/Tooltip/Tooltip.js (hunk 5 above)
Comment (cristianoliveira) [reply src/components/Tooltip/Tooltip.js]: Yeah. Connor told me about this pattern on the last PR :) 
Let me change

File: src/components/Tooltip/Tooltip.js (hunk 6 above)
Comment (cristianoliveira) [reply src/components/Tooltip/Tooltip.js]: Sure thing :)

File: src/components/Tooltip/Tooltip.js (hunk 9 above)
Comment (cristianoliveira) [reply src/components/Tooltip/Tooltip.js]: Do u mean like declaring them at the beginning?

File: src/components/Tooltip/Tooltip.spec.js [hunk 11]
```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+
+import Tooltip from '.';
+
+describe('Tooltip', () => {
+  it('aligns to the center', () => {
+    const component = create(
+      <Tooltip align={Tooltip.Center} content="The tooltip content">
+        Something with tooltip
```
[the remark this answers is not in the document]
Comment (cristianoliveira) [reply src/components/Tooltip/Tooltip.spec.js]: Added one for it and one for rendering icons as tooltip `content` also added some examples on `storybook`

**Pull request merged**